kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/kms/vault/KubeVaultTestKmsFacade.java

```diff
@@ -53,14 +53,20 @@ public class KubeVaultTestKmsFacade extends AbstractVaultTestKmsFacade {
     /**
      * Instantiates a new Kube vault test kms facade.
      *
-     * @param namespace the namespace
-     * @param podName the pod name
-     * @param openshiftCluster the boolean for openshift cluster
      */
-    public KubeVaultTestKmsFacade(String namespace, String podName, boolean openshiftCluster) {
-        this.namespace = namespace;
-        this.podName = podName;
-        this.vault = new Vault(namespace, VAULT_ROOT_TOKEN, openshiftCluster);
+    public KubeVaultTestKmsFacade() {
+        this.namespace = Vault.VAULT_DEFAULT_NAMESPACE;
+        this.podName = Vault.VAULT_POD_NAME;
+        this.vault = new Vault(VAULT_ROOT_TOKEN);
+    }
+
+    /**
+     * Sets openshift cluster.
+     *
+     * @param openshiftCluster the openshift cluster
+     */
+    public void setOpenshiftCluster(boolean openshiftCluster) {
+        this.vault.setOpenshiftCluster(openshiftCluster);
```

Comment:
can't we just discover if we are running on OpenShift?  The technique I have seen us use many times is to query for OpenShift Routes.  If you get a result (or empty list), you are on openshift.  Otherwise you are on Kubernetes.

Reply:
Something like `kubectl api-resources --api-group=route.openshift.io`